src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs

```diff
@@ -20,17 +21,28 @@ public static class DbContextExtensions
         /// Ensures the automatic history.
         /// </summary>
         /// <param name="context">The context.</param>
-        public static void EnsureAutoHistory(this DbContext context)
+        /// <param name="useAddState">EntityStatus.Added will user too</param>
+        public static void EnsureAutoHistory(this DbContext context, bool useAddState = false)
         {
-            EnsureAutoHistory<AutoHistory>(context, () => new AutoHistory());
+            EnsureAutoHistory<AutoHistory>(context, () => new AutoHistory(), useAddState);
         }
 
-        public static void EnsureAutoHistory<TAutoHistory>(this DbContext context, Func<TAutoHistory> createHistoryFactory)
+        public static void EnsureAutoHistory<TAutoHistory>(this DbContext context, Func<TAutoHistory> createHistoryFactory, bool useAddState = false)
             where TAutoHistory : AutoHistory
         {
             // Must ToArray() here for excluding the AutoHistory model.
             // Currently, only support Modified and Deleted entity.
-            var entries = context.ChangeTracker.Entries().Where(e => e.State == EntityState.Modified || e.State == EntityState.Deleted).ToArray();
+            var allEntries = context.ChangeTracker.Entries();
```

Comment:
Maybe the user needs only deleted items to be logged.
```suggestion
            var allEntries = context.ChangeTracker.Entries().Where(e => e.State !=Detached && e.State != Unchanged && predicate(e.State));
```